print-ldp.c

```diff
@@ -491,17 +522,21 @@ ldp_tlv_print(netdissect_options *ndo,
 	break;
 
     case LDP_TLV_STATUS:
-	TLV_TCHECK(8);
+	TLV_TCHECK(10);
 	ui = GET_BE_U_4(tptr);
 	tptr+=4;
-	ND_PRINT("\n\t      Status: 0x%02x, Flags: [%s and %s forward]",
-	       ui&0x3fffffff,
-	       ui&0x80000000 ? "Fatal error" : "Advisory Notification",
-	       ui&0x40000000 ? "do" : "don't");
+	ND_PRINT("\n\t      Status Code: %s, Flags: [%s and %s forward]",
+		 tok2str(ldp_status_code_values, "Unknown", ui&0x3fffffff),
+		 ui&0x80000000 ? "Fatal error" : "Advisory Notification",
+		 ui&0x40000000 ? "do" : "don't");
 	ui = GET_BE_U_4(tptr);
 	tptr+=4;
 	if (ui)
 	    ND_PRINT(", causing Message ID: 0x%08x", ui);
+	ui = GET_BE_U_2(tptr);
+	if (ui)
+	    ND_PRINT(", Message ID: %s", tok2str(ldp_msg_values, "Unknown", ui));
```

Comment:
Section 3.4.6 of RFC 5036 calls this 16-bit field "Message Type".

Reply:
my guess is that ui just means unsigned integer and is used as a general purpose variable for extracting.
ui is being used for extracting status code and message_id previously - shall i change that as well ?

Reply:
The variable name is not a problem. The label is wrong:
```
    0                   1                   2                   3
    0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
   |U|F| Status (0x0300)           |      Length                   |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
   |                     Status Code                               |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
   |                     Message ID                                |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
   |      Message Type             |
   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
```
What the new code prints is "Message Type", and "Message ID" was printed just before.

Reply:
got it - nice catch;